File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int left, int right) {
    int answer = 0;

    for(int i = left ; i <= right; i++){
        int cnt = 0;
        for (int d = 1 ;d*d <= i; d ++){
            if (d*d == i){
                cnt ++;
            }
            if (i%d == 0){
                cnt += 2;
            }
        }
        if (cnt%2){
            answer -= i;
        }else{
            answer += i;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0
    
    for num in range(left,right+1):
        d = 1
        cnt = 0
        while(d**2<=num):
            if d**2 == num:
                cnt += 1
            if num%d == 0:
                cnt += 2
            d += 1
        if cnt%2:
            answer -= num
        else:
            answer += num
            
    
    return answer